Moved some examples to new /examples route

damnko opened the pull request:
Hi, I created a new "Examples" route with some subpages with all the examples divided by context.
This should hopefully solve #184

I wanted to place all the examples in a separate module and load it asynchronously but I kept having problems with Webpack.
These are the `loadChildren` test codes I used with the `angular2-load-children-loader` Webpack loader
```
loadChildren: () => require("./examples/examples.module")("ExamplesModule")
loadChildren: 'es6-promise?,[name]!./examples/examples.module#ExamplesModule'
loadChildren: 'es6-promise!./examples/examples.module#ExamplesModule'
```
None of these worked and Webpack kept throwing the following error:
`Module build failed: TypeError: Can't add property q9ybxdsfw29, object is not extensible`

If you have any hint on how to solve this please let me know. Thanks.

LexZhukov commented on demo/src/app/pages/examples/examples/custom-edit-view-examples.html:
```diff
@@ -0,0 +1,25 @@
+<h2>Customized edit and view cells examples</h2>
```
Please, for file naming use angular style guide. For example: `custom-edit-view-examples.component.ts`, `custom-edit-view-examples.component.html`, `custom-edit-view-examples.component.scss`, `custom-edit-view-examples.service.ts`.

LexZhukov commented:
@damnko how to implement async routes you can see in ng2-admin repository. PR looks good 😃 .

damnko commented:
Hi, I have moved the /examples section to a dedicated async module but I am still having problems when running `npm run build:aot`, I am getting this error
```
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at afterInnerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:138:10)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at Resolver.applyPluginsAsyncSeriesBailResult1 (/ng2-smart-table/node_modules/tapable/lib/Tapable.js:181:46)
    at innerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:125:19)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at /ng2-smart-table/node_modules/tapable/lib/Tapable.js:283:15
    at innerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:123:11)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at /ng2-smart-table/node_modules/tapable/lib/Tapable.js:283:15
    at resolver.doResolve.createInnerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:44:6)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at afterInnerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:138:10)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at next (/ng2-smart-table/node_modules/tapable/lib/Tapable.js:188:11)
    at innerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:123:11)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at /ng2-smart-table/node_modules/tapable/lib/Tapable.js:283:15
    at resolver.doResolve.createInnerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:44:6)
    at loggingCallbackWrapper (/ng2-smart-table/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at afterInnerCallback (/ng2-smart-table/node_modules/enhanced-resolve/lib/Resolver.js:138:10)
 @ ./demo/src/main.browser.aot.ts 11:29-88
```

When doing `npm start` everything seems to work correctly.

Could you give it a try and see if you come up with anything that can solve this issue?

I fixed all the suggestions you wrote me on the previous commit apart from changing the `routerLink` from absolute to relative because by removing the `/` it appends the url each time (eg. by clicking _demo_ first and then _documentation_ you get  `/demo/documentation`).

Please let me know should you have any question or suggestion

LexZhukov commented:
@damnko for a quick fix, replace async routes with components and we will ready to merge it.

nnixaa commented:
Hey @damnko, really looking forward to your input on this :)

damnko commented:
Hi, I'm sorry for not being responsive.
The fact is my first PR had _sync_ routing, then I moved to the async version and then I was asked to roll back to _sync_ routing again. I preferred to have a feedback on how to solve the problem rather then bypassing it.

nnixaa commented:
Hey @damnko, thanks for your response. Well, in this very case, we decided that at the moment it's much more important to have the examples pages working properly rather than have the async version of the routes :)
Anyway, we already merged the PR and @LexZhukov will complete the integration, then we can see if it makes sense to proceed with the async version or leave it for now.

Thanks for your work!

damnko commented:
Sure I understand, thanks for your feedback.
